### PR TITLE
SyntaxTree: always keep parsed implicit constructors

### PR DESCRIPTION
**Pull request by auduchinok**
Keeps parsed implicit constructors for types that are not allowed to have them in the syntax tree.

### CODE REVIEW

**Comment by auduchinok** [comment]
It's ready.

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
Thanks!

File: src/fsharp/service/ServiceUntypedParse.fs [hunk 1]
```diff
@@ -600,7 +600,7 @@ type FSharpParseFileResults(errors: FSharpDiagnostic[], input: ParsedInput optio
                           yield! walkExpr true e ]
             
             // Process a class declaration or F# type declaration
-            let rec walkTycon (TypeDefn(ComponentInfo(_, _, _, _, _, _, _, _), repr, membDefns, m)) =
+            let rec walkTycon (TypeDefn(ComponentInfo(_, _, _, _, _, _, _, _), repr, membDefns, _, m)) =
```
Comment (dsyme) [inline src/fsharp/service/ServiceUntypedParse.fs:603]: This should walk the implicit ctor if it's there

File: src/fsharp/service/ServiceUntypedParse.fs [hunk 2]
```diff
@@ -1236,7 +1236,7 @@ module UntypedParseImpl =
             | SynTypeDefnSigRepr.Simple(defn, _) -> walkTypeDefnSimple defn
             | SynTypeDefnSigRepr.Exception(_) -> None
 
-        and walkTypeDefn (TypeDefn (info, repr, members, _)) =
+        and walkTypeDefn (TypeDefn (info, repr, members, _, _)) =
```
Comment (dsyme) [inline src/fsharp/service/ServiceUntypedParse.fs:1239]: This should walk the implicit ctor if it's there

File: src/fsharp/service/ServiceNavigation.fs [hunk 3]
```diff
@@ -173,7 +173,7 @@ module NavigationImpl =
             let nested = processMembers membDefns FSharpEnclosingEntityKind.Exception |> snd
             processExnDefnRepr baseName nested repr
 
-        and processTycon baseName (TypeDefn(ComponentInfo(_, _, _, lid, _, _, access, _), repr, membDefns, m)) =
+        and processTycon baseName (TypeDefn(ComponentInfo(_, _, _, lid, _, _, access, _), repr, membDefns, _, m)) =
```
Comment (dsyme) [inline src/fsharp/service/ServiceNavigation.fs:176]: This should walk the implicit ctor if it's there?

File: src/fsharp/service/ServiceNavigation.fs [hunk 4]
```diff
@@ -686,7 +686,7 @@ module NavigateTo =
             | SynModuleDecl.HashDirective _
             | SynModuleDecl.Open _ -> ()
     
-        and walkSynTypeDefn(TypeDefn(componentInfo, representation, members, _)) container = 
+        and walkSynTypeDefn(TypeDefn(componentInfo, representation, members, _, _)) container =
```
Comment (dsyme) [inline src/fsharp/service/ServiceNavigation.fs:689]: This should walk the implicit ctor if it's there

File: src/fsharp/service/ServiceInterfaceStubGenerator.fs [hunk 5]
```diff
@@ -702,7 +702,7 @@ module InterfaceStubGenerator =
                 | SynModuleDecl.Open _ -> 
                     None
 
-        and walkSynTypeDefn(TypeDefn(_componentInfo, representation, members, range)) = 
+        and walkSynTypeDefn(TypeDefn(_componentInfo, representation, members, _, range)) =
```
Comment (dsyme) [inline src/fsharp/service/ServiceInterfaceStubGenerator.fs:705]: This should walk the implicit ctor if it's there

File: src/fsharp/service/ServiceAssemblyContent.fs [hunk 6]
```diff
@@ -753,7 +753,7 @@ module ParsedInput =
             | SynTypeDefnSigRepr.Simple(defn, _) -> walkTypeDefnSimple defn
             | SynTypeDefnSigRepr.Exception _ -> ()
     
-        and walkTypeDefn (TypeDefn (info, repr, members, _)) =
+        and walkTypeDefn (TypeDefn (info, repr, members, _, _)) =
```
Comment (dsyme) [inline src/fsharp/service/ServiceAssemblyContent.fs:756]: This should walk the implicit ctor if it's there

File: src/fsharp/service/ServiceUntypedParse.fs (hunk 1 above)
Comment (auduchinok) [reply src/fsharp/service/ServiceUntypedParse.fs:603]: I expect it to walk in it for object model types if it already does, this behaviour hasn't changed. Do you mean, it should walk in it for types that don't allow implicit constructors too?  This PR only adds thrown constructors back to non-object model types.

**Pull request merged**